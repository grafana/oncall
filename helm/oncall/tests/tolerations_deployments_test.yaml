suite: test image and imagePullPolicy for deployments
templates:
  - celery/deployment.yaml
  - engine/deployment.yaml
  - engine/job-migrate.yaml
release:
  name: oncall
tests:
  - it: tolerations={} -> should set tolerations null
    asserts:
      - notExists:
          path: spec.template.spec.tolerations

  - it: tolerations -> should use custom tolerations
    set:
      migrate:
        tolerations:
          - key: "node-role.kubernetes.io/unittest"
            operator: "Exists"
            effect: "NoSchedule"
      engine:
        tolerations:
          - key: "node-role.kubernetes.io/unittest"
            operator: "Exists"
            effect: "NoSchedule"
      celery:
        tolerations:
          - key: "node-role.kubernetes.io/unittest"
            operator: "Exists"
            effect: "NoSchedule"
    asserts:
      - matchSnapshot:
          path: spec.template.spec.tolerations
