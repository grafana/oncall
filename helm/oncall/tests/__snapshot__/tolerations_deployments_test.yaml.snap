tolerations -> should use custom tolerations:
  1: |
    - effect: NoSchedule
      key: node-role.kubernetes.io/unittest
      operator: Exists
  2: |
    - effect: NoSchedule
      key: node-role.kubernetes.io/unittest
      operator: Exists
  3: |
    - effect: NoSchedule
      key: node-role.kubernetes.io/unittest
      operator: Exists
